src/nupic/algorithms/SDRClassifier.hpp

```diff
@@ -22,203 +22,213 @@
  * --------------------------------------------------------------------- */
 
 /** @file
- * Definitions for the SDRClassifier.
+ * Definitions for the SDR Classifier & Predictor.
  */
 
 #ifndef NTA_SDR_CLASSIFIER_HPP
 #define NTA_SDR_CLASSIFIER_HPP
 
 #include <deque>
-#include <iostream>
 #include <map>
-#include <string>
 #include <vector>
 
 #include <nupic/types/Types.hpp>
+#include <nupic/types/Sdr.hpp>
 #include <nupic/types/Serializable.hpp>
 
 namespace nupic {
 namespace algorithms {
 namespace sdr_classifier {
 
 
-const UInt sdrClassifierVersion = 2;
-
 /**
- * PDF - Probability Density Function, distribution of likelihood of values
+ * PDF - Probability Distribution Function, distribution of likelihood of values
+ *       for each category.
+ *
+ * See also:  https://en.wikipedia.org/wiki/Probability_distribution
  */
-using PDF = std::vector<Real64>;
+using PDF = std::vector<Real>;
 
 /**
- * The key is the step, for predicting multiple time steps into the future.
- * The key ACTUAL_VALUES contains an estimate of the actual values.
- * The value is a PDF(probability density function, list of probabilities of outcomes) 
- * of the result being in each bucket.
+ * Returns the class with the single greatest probablility.
  */
-const Int ACTUAL_VALUES = -1;
-using ClassifierResult = std::map<Int, PDF>;
+UInt getClassification( const PDF & data );
 
 
-class SDRClassifier : public Serializable
+/**
+ * The SDR Classifier takes the form of a single layer classification network
+ * that takes SDRs as input and outputs a predicted distribution of classes.
```

Comment:
high-level: this whole classifier is not a part of HTM, can we find a simple 3rd party NN library to delegate the task to? And just have a wrapper to call it with SDR and input?

Reply:
I'd rather not:
- This was already implemented & tested by numenta.
- This works pretty well. I don't think a NN library would do significantly better than this.
- This is a very simple network, I think a whole NN framework is overkill.

Reply:
> This works pretty well. I don't think a NN library would do significantly better than this. 

well, this is a room for experimentation. But ok, not concern of this PR